docs: Update development setup with preferred kind-based approach (#25535)
cilium-team/cilium: Documentation/contributing/development/dev_setup.rst

VirtualBox on Linux, but `Docker for Mac
<https://docs.docker.com/desktop/install/mac-install/>`_ is required for Mac
OS.

There are Makefile targets that make it easy to use to spin up an environment
and build Cilium images.

* ``make kind``: Creates a kind cluster based on configuration passed in (see
  below for configuration).
* ``make kind-image``: Builds all Cilium images and loads them into the
  cluster.
* ``make kind-image-agent``: Builds the Cilium Agent image only and loads it
  into the cluster.
* ``make kind-image-operator``: Builds the Cilium Operator (generic) image only
  and loads it into the cluster.
* ``make kind-image-debug``: Builds all Cilium images with optimizations
  disabled and ``dlv`` embedded for live debugging enabled and loads the images
  into the cluster.
* ``make kind-install-cilium``: Installs Cilium into the cluster using the
  Cilium CLI.
* ``make kind-down``: Tears down and deletes the cluster.

The above is a non-exhaustive list, yet includes the most used commands for
convenience. See the Makefile for more specific targets.

Configuration for clusters
^^^^^^^^^^^^^^^^^^^^^^^^^^

> TODO

Reviewer: this is what I do https://gist.github.com/aojea/22a772dcf51948d955ac7cc3b75e7776

the env variables for creating the cluster are important at least to me, since I need to replcae kube-proxy
```
KUBEPROXY_MODE="none" make kind
```

Author: Thanks, updated.

| `VirtualBox <https://www.virtualbox.org/wiki/Downloads>`_  | >= 5.2                | N/A (OS-specific)                                                              |
+------------------------------------------------------------+-----------------------+--------------------------------------------------------------------------------+

Kind-based Setup (preferred)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The setup for a `kind <https://kind.sigs.k8s.io/>`_ environment lives in
> ``contrib/scripts/kind.sh``. This setup does not require any VMs and/or
> VirtualBox on Linux, but `Docker for Mac
> <https://docs.docker.com/desktop/install/mac-install/>`_ is required for Mac
> OS.

Reviewer 2: Use active voice, avoid idioms
```suggestion
You can find the setup for a `kind <https://kind.sigs.k8s.io/>`_ environment in
``contrib/scripts/kind.sh``. This setup doesn't require any VMs and/or
VirtualBox on Linux, but does require `Docker for Mac
<https://docs.docker.com/desktop/install/mac-install/>`_ for Mac
OS.
```

> There are Makefile targets that make it easy to use to spin up an environment
> and build Cilium images.

Reviewer 2: Avoid words like "easy", "just", "simply"
```suggestion
Makefile targets automate the task of spinning up an environment
and building Cilium images:
```

> * ``make kind``: Creates a kind cluster based on configuration passed in (see
>   below for configuration).

Reviewer 2: s/below/following/g, consistent format: "for more information about `x`, see `y`"
```suggestion
* ``make kind``: Creates a kind cluster based on the configuration passed in. For more information, see _`Configuration for clusters`.
```

* ``make kind-image``: Builds all Cilium images and loads them into the
  cluster.
* ``make kind-image-agent``: Builds the Cilium Agent image only and loads it
  into the cluster.
* ``make kind-image-operator``: Builds the Cilium Operator (generic) image only
  and loads it into the cluster.
* ``make kind-image-debug``: Builds all Cilium images with optimizations
  disabled and ``dlv`` embedded for live debugging enabled and loads the images
  into the cluster.
* ``make kind-install-cilium``: Installs Cilium into the cluster using the
  Cilium CLI.
* ``make kind-down``: Tears down and deletes the cluster.

> The above is a non-exhaustive list, yet includes the most used commands for
> convenience. See the Makefile for more specific targets.

Reviewer 2: s/above/preceding/g, use clear, plain language
```suggestion
The preceding list includes the most used commands for
convenience. For more targets, see the `Makefile`.
```

Configuration for clusters
^^^^^^^^^^^^^^^^^^^^^^^^^^

``make kind`` takes a few environment variables to modify the configuration of

> the clusters it creates. The following are the most commonly used parameters.

Reviewer 2: Clarity
```suggestion
the clusters it creates. The following parameters are the most commonly used:
```

* ``CONTROLPLANES``: How many control-plane nodes are created.
* ``WORKERS``: How many worker nodes are created.
* ``CLUSTER_NAME``: The name of the Kubernetes cluster.

> * ``IMAGE``: The image for kind, e.g. ``kindest/node:v1.11.10``

Reviewer 2: End list items consistently with or without periods, no Latin abbreviations
```suggestion
* ``IMAGE``: The image for kind, for example: ``kindest/node:v1.11.10``.
```

* ``KUBEPROXY_MODE``: Pass directly as ``kubeProxyMode`` to the kind

>   configuration CRD.

Reviewer 2: Define acronyms on first use on page
```suggestion
  configuration Custom Resource Definition (CRD).
```

> To see the rest, please see ``contrib/scripts/kind.sh``.

Reviewer 2: Use clear, plain language, consistent format
```suggestion
For more environment variables, see ``contrib/scripts/kind.sh``.
```
